apps/homework/tasks.py

```diff
@@ -14,41 +15,63 @@
 
 logger = logging.getLogger(__name__)
 
+
 class SubmissionPostException(BaseException):
 
     def __init__(self, *args, **kwargs):
         self.sub_type = kwargs.get('sub_type')
         super(SubmissionPostException, self).__init__(*args, **kwargs)
 
+
+def retrieve_score_from_jupyter_notebook(fd):
+    content = fd.readlines()
+    filtered_content = list(filter(lambda line: 'final score' in str(line) and 'print' not in str(line), content))
+    if len(filtered_content) is not 1:
+        raise IndexError('Exactly one line should contain the phrase "final score" and not include the phrase "print".')
+    score_string = str(filtered_content[0])
+    start_index = score_string.find('is')
+
+    found = re.search('[0-9]+ \/ [0-9]+', score_string)
+
+    # Check for no matches
+    separated = found.group().split(' ')
+    numerator = int(separated[0])
+    denominator = int(separated[2])
+    return numerator, denominator
+
+
 @task
 def post_submission(submission_pk, data_file=None):
     # Get our URL's formatted and such
     submission = Submission.objects.get(pk=submission_pk)
-    parsed_uri = urlparse(submission.get_challenge_url)
-    scheme = parsed_uri.scheme
-    domain = parsed_uri.netloc
-    path = parsed_uri.path
-    challenge_pk = path.split('/')[-1]
-    site_url = "{0}://{1}".format(scheme, domain)
-    submission_url = '{0}/api/competition/{1}/submission/sas'.format(site_url, challenge_pk)
-
-    logger.info(f"Making submission to {submission_url}")
-
-    # Post our request to the submission SAS API endpoint
-    resp = requests.post(
-        url=submission_url, auth=HTTPBasicAuth(
-            os.environ.get('CODALAB_SUBMISSION_USERNAME'),
-            os.environ.get('CODALAB_SUBMISSION_PASSWORD')
+    submission_data = None
+    s3_file_url = None
+
+    if not submission.definition.jupyter_notebook_enabled:
+        parsed_uri = urlparse(submission.get_challenge_url)
+        scheme = parsed_uri.scheme
+        domain = parsed_uri.netloc
+        path = parsed_uri.path
+        challenge_pk = path.split('/')[-1]
+        site_url = "{0}://{1}".format(scheme, domain)
+        submission_url = '{0}/api/competition/{1}/submission/sas'.format(site_url, challenge_pk)
+
+        logger.info(f"Making submission to {submission_url}")
+
+        # Post our request to the submission SAS API endpoint
+        resp = requests.post(
+            url=submission_url, auth=HTTPBasicAuth(
+                os.environ.get('CODALAB_SUBMISSION_USERNAME'),
```

Comment:
we should probably be reading these into `settings/base.py` and doing `from django.conf import settings` `settings.CODALAB_SUBMISSION_USERNAME`

Reply:
✅ Good call!